receiver/vmmetricsreceiver/metrics_receiver.go

```diff
@@ -85,51 +44,20 @@ func (vmr *Receiver) MetricsSource() string {
 
 // StartMetricsReception scrapes VM metrics based on the OS platform.
 func (vmr *Receiver) StartMetricsReception(host receiver.Host) error {
-	vmr.mu.Lock()
```

Comment:
I guess this lock was needed to avoid concurrent execution of StartMetricsReception and StopMetricsReception. Is this unnecessary?

Reply:
That makes sense - put it back.